service/history/replication/executable_task_converter.go

```diff
@@ -93,45 +100,87 @@ func (e *executableTaskConverterImpl) convertOne(
 			replicationTask.SourceTaskId,
 			taskCreationTime,
 			taskClusterName,
-		)
+		), nil
 	case enumsspb.REPLICATION_TASK_TYPE_HISTORY_METADATA_TASK: // TODO to be deprecated
 		return NewExecutableNoopTask(
 			e.processToolBox,
 			replicationTask.SourceTaskId,
 			taskCreationTime,
 			taskClusterName,
-		)
+		), nil
 	case enumsspb.REPLICATION_TASK_TYPE_SYNC_ACTIVITY_TASK:
 		return NewExecutableActivityStateTask(
 			e.processToolBox,
 			replicationTask.SourceTaskId,
 			taskCreationTime,
 			replicationTask.GetSyncActivityTaskAttributes(),
 			taskClusterName,
-		)
+		), nil
 	case enumsspb.REPLICATION_TASK_TYPE_SYNC_WORKFLOW_STATE_TASK:
 		return NewExecutableWorkflowStateTask(
 			e.processToolBox,
 			replicationTask.SourceTaskId,
 			taskCreationTime,
 			replicationTask.GetSyncWorkflowStateTaskAttributes(),
 			taskClusterName,
-		)
+		), nil
 	case enumsspb.REPLICATION_TASK_TYPE_HISTORY_V2_TASK:
+		events, newRunEvents, err := e.deserializeHistoryEvents(replicationTask.SourceTaskId, replicationTask.GetHistoryTaskAttributes())
+		if err == nil {
+			return nil, err
+		}
+		if len(events) == 0 {
+			e.processToolBox.Logger.Error("unable to create history replication task, no events",
+				tag.WorkflowNamespaceID(replicationTask.GetHistoryTaskAttributes().GetNamespaceId()),
+				tag.WorkflowID(replicationTask.GetHistoryTaskAttributes().GetWorkflowId()),
+				tag.WorkflowRunID(replicationTask.GetHistoryTaskAttributes().GetRunId()),
+				tag.TaskID(replicationTask.SourceTaskId),
+			)
+			return nil, serviceerror.NewInvalidArgument("unable to create replication task, no events")
+		}
```

Comment:
this logic should be moved into within task itself, so we can put this task into DLQ